include/event_manager.rst

```diff
@@ -0,0 +1,247 @@
+.. _event_manager:
+
+Event Manager
+#############
+
+Overview
+========
+
+Event Manager module provides an infrastructure required for the implementation using an event-driven architecture approach.
+Modules (listeners) communicate and transfer data using events, which are processed by Event Manager.
+
+Core features:
+**************
+
+* Event-based communication with **multiple listeners** for given event type.
+* Events can be used **instead of direct function calls** (reduces number of direct dependencies between modules).
+* New event types or modules may be easily and quickly added to application.
+* Modules may be easily swapped - useful when creating multiple configurations (e. g. application working with various boards).
+* **Dedicated profiler** enables observing events propagation in the system, data connected with events or creating stats.
+* **Shell integration** for displaying additional information, dynamically enabling/disabling logging for given event types.
+
+The sample is available at :ref:`event_manager_sample`.
+
+Required configuration
+=================================
+
+* :option:`CONFIG_LINKER_ORPHAN_SECTION_PLACE` - Event Manager uses orphan memory sections
+* :option:`CONFIG_HEAP_MEM_POOL_SIZE` - events are dynamically allocated using heap memory
```

Comment:
What should this option be set to? And why?

Reply:
we are using dynamic memory allocation for events, dynamic memory allocation allocates data on heap so we need a proper heap size (big enough to allocate the events - exact value depens on application)